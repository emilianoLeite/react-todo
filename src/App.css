.App {
  display: flex;
  flex: 1;
  align-items: center;
  flex-direction: column
}

.completed {
  font-style: italic;
  text-decoration: line-through;
}

.todo-list li {
 list-style-type: none;
 margin-left: -19px;
}
