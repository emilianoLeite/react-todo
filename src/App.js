import React, { Component } from 'react';
import './App.css';

import TodoInput from './components/todo-input';
import TodoList from './components/todo-list';
import Todo from './models/todo';

export default class App extends Component {
  constructor(props) {
    super(props);
    this.state = { todos: this.getTodos() };
    this.setTodos = this.setTodos.bind(this);
    this.updateTodo = this.updateTodo.bind(this);
    this.commitChanges = this.commitChanges.bind(this);
  }

  getTodos() {
    const todosProps = JSON.parse(sessionStorage.getItem('todos-list'));

    if (todosProps !== null) {
      return Todo.wrap(todosProps);
    } else {
      return [];
    }
  }

  setTodos(todoText){
    const newTodo = new Todo({ text: todoText });
    const todos = this.state.todos.concat(newTodo);
    this.commitChanges(todos);
  }

  updateTodo(todoIndex){
    const todos = this.state.todos;
    todos[todoIndex] = todos[todoIndex].toggleCompletion();
    this.commitChanges(todos);
  }

  commitChanges(todos) {
    this.setState({ todos });
    sessionStorage.setItem('todos-list', JSON.stringify(todos));
  }

  render() { return (
    <div className="App">
      <TodoInput onCreateTodo={this.setTodos}/>
      <TodoList
        onToggledTodo={this.updateTodo}
        todos={this.state.todos}
      />
    </div>
  );
  }
}
