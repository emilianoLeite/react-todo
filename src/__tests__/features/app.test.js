import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import App from '../../App';

let subject;

beforeEach(() => {
  subject = mount(<App />);
});

afterEach(() => {
  subject.unmount();
});

const createNewTodo = (todoText) => {
  subject.find('TodoInput').simulate('keyPress', { target: { value: todoText }, keyCode: 13 });
};

it('allows the creation and listing of todos', () => {
  let currentSnapshot = renderer
    .create(subject)
    .toJSON();
  expect(currentSnapshot).toMatchSnapshot(); // No TODOs

  createNewTodo('myNewTodo');

  currentSnapshot = renderer
    .create(subject)
    .toJSON();
  expect(currentSnapshot).toMatchSnapshot(); // Shows created TODO

  createNewTodo('myNewTodo2');

  currentSnapshot = renderer
    .create(subject)
    .toJSON();
  expect(currentSnapshot).toMatchSnapshot(); // Shows all TODOs
});

