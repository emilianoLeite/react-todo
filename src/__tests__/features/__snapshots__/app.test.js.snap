// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`allows the creation and listing of todos 1`] = `
<div
  className="App"
>
  <input
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="New todo..."
    type="text"
    value=""
  />
  <ul
    className="todo-list"
  />
</div>
`;

exports[`allows the creation and listing of todos 2`] = `
<div
  className="App"
>
  <input
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="New todo..."
    type="text"
    value=""
  />
  <ul
    className="todo-list"
  >
    <li
      className=""
    >
      <input
        onClick={[Function]}
        type="checkbox"
      />
      myNewTodo
    </li>
  </ul>
</div>
`;

exports[`allows the creation and listing of todos 3`] = `
<div
  className="App"
>
  <input
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="New todo..."
    type="text"
    value=""
  />
  <ul
    className="todo-list"
  >
    <li
      className=""
    >
      <input
        onClick={[Function]}
        type="checkbox"
      />
      myNewTodo
    </li>
    <li
      className=""
    >
      <input
        onClick={[Function]}
        type="checkbox"
      />
      myNewTodo2
    </li>
  </ul>
</div>
`;
